File: code/php/delete1.php
<?php
require_once("xapian.php");

// Start of example code.
function delete_docs ($dbpath, $identifiers)
{
    // Open the database we're going to be deleting from.
    $db = new XapianWritableDatabase($dbpath, Xapian::DB_OPEN);

    foreach ($identifiers as $identifier)
    {
        $idterm = 'Q' . $identifier;
        $db->delete_document($idterm);
    }
}
// End of example code.

if ($argc < 3) {
    print "Usage: php $argv[0] DBPATH ID...\n";
    die();
}

// Call the delete_docs function.
delete_docs($argv[1], array_slice($argv, 2));
?>

File: code/ruby/delete1.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'xapian'

### Start of example code.
def delete_docs(dbpath, identifiers)
  db = Xapian::WritableDatabase.new(dbpath, Xapian::DB_OPEN)
  identifiers.each do |identifier|
    idterm = "Q#{identifier}"
    db.delete_document(idterm)
  end
end
### End of example code.

abort "Usage #{__FILE__} DBPATH ID..." if ARGV.length < 2

delete_docs(ARGV[0], ARGV[1..])
